src/python/pants/backend/python/rules/run_setup_py.py

```diff
@@ -194,44 +199,69 @@ def register_options(cls, register):
            "`--setup-py2-args=\"bdist_wheel --python-tag py36.py37\"`. If unspecified, we just "
            "dump the setup.py chroot."
     )
+    register(
+      '--recursive', type=bool,
+      help="If specified, will run the setup.py command recursively on all exported targets that "
+           "the specified targets depend on, in dependency order.  This is useful, e.g., when "
+           "the command publishes dists, to ensure that any dependencies of a dist are published "
+           "before it."
+    )
 
 
 class SetupPy(Goal):
   subsystem_cls = SetupPyOptions
 
 
+def validate_args(args: Tuple[str, ...]):
+  # We rely on the dist dir being the default, so we know where to find the created dists.
+  if '--dist-dir' in args or '-d' in args:
+    raise InvalidSetupPyArgs('You must not set --dist-dir in your setup.py args')
```

Comment:
It would help to follow this up with where users can expect the default dist dir to be.

Reply:
It looks like this is `distdir.relpath`? It would be great to dynamically compute this path and display in this error message, but at a minimum pointing to the option they can configure would help.

Reply:
Users don't need to know about this distdir - this is setuptool's internal concept, we rake up the files from under there and plop them in our distdir. This is just defense against the user changing what we expect for some reason.

Reply:
Yes, agreed. But presumably they are trying to set the option `--dist-dir` here because they care about where this all ends up getting written. Currently, the error message only says "You can't do this."

Instead, I find it more helpful to say "You can't do this directly through passthrough args, but you can use `./pants --distdir` to get the same effect."